plugins/auth-backend/src/identity/IdentityClient.ts

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import fetch from 'cross-fetch';
+import { JWKECKey } from 'jose';
+import { PluginEndpointDiscovery } from '@backstage/backend-common';
+
+/**
+ * A identity client to interact with auth-backend.
+ */
+export class IdentityClient {
+  private readonly discovery: PluginEndpointDiscovery;
+
+  constructor(options: { discovery: PluginEndpointDiscovery }) {
+    this.discovery = options.discovery;
+  }
+
+  /**
+   * Lists public part of keys used to sign Backstage Identity tokens
+   */
+  async listPublicKeys(): Promise<{
+    keys: JWKECKey[];
+  }> {
+    const url = `${await this.discovery.getBaseUrl(
+      'auth',
```

Comment:
I am not sure that this is the right endpoint. 
Authentication providers are normally hosted under `/auth/provider/` and therefore any request to the `/.well-known` endpoint should also be made to `/auth/provider/.well-know/jwks.json`

Reply:
I am fairly confident this is right, and it does work for me locally. See https://github.com/erikxiv/backstage/blob/feat/api-auth-backend/plugins/auth-backend/src/service/router.ts line 133 vs 140, where the latter adds the provider path element.

Reply:
Yep, it's correct 👍